Linux_Install.sh

```diff
@@ -0,0 +1,227 @@
+	echo "        ulocal  - uninstall IDE from user $USER"
+	echo "usage: sudo ./Linux_Install.sh [system] | [usystem]"	
+	echo "        system  - install IDE systemwide for all users"
+	echo "        usystem - uninstall IDE systemwide from all users\n"
+	echo "Both local and system installations may be used on the same computer"
+	echo "for example - a stable release installed systemwide and a nightly for a development user\n"
+
+	MSG="Install IDE 2.0 for user $USER only?"
+
+elif [ $1 = local ]
+then
+		MSG="Install IDE 2.0 for user $USER only?"
+elif [ $1 = ulocal ]
+then
+		MSG="UnInstall IDE 2.0 from user $USER?"
+		MODE=u
+elif [ $1 = usystem ]
+then
+		MSG="UnInstall IDE 2.0 system wide (all users will be affected)?"
+		MODE=s
+elif [ $1 = system ]
+then
+		MSG="Install IDE 2.0 system wide (all users will have access)?"
+		MODE=S
+fi
+	
+# Get absolute path from which this script file was executed
+# (Could be changed to "pwd -P" to resolve symlinks to their target)
+SCRIPT_PATH=$( pwd -P )
+LIB_PATH=$SCRIPT_PATH
+EXE_PATH=$SCRIPT_PATH
+# echo S_PATH = $SCRIPT_PATH
+
+
+read -p "$MSG (Y/N) " YN
+
+if [ -z $YN ]
+then
+	echo OK - exiting
+	exit	
+elif [ $YN = n ]
+then
+	echo OK - No is No... exiting
+	exit
+fi	
+
+echo
+
+###### Remove local user installation
+	
+	if [ $MODE = u ]
+	then
+		echo Uninstalling IDE 2.0 from user $USER
+		echo "Deleting ${HOME}/.local/share/applications/${RESOURCE_NAME}l.desktop"
+		rm   ${HOME}/.local/share/applications/${RESOURCE_NAME}l.desktop
+		
+		echo "Deleting ${HOME}/.local/lib/${RESOURCE_NAME}"
+		rm -rf ${HOME}/.local/lib/${RESOURCE_NAME}
+		
+		echo "Deleting ${HOME}/.local/bin/${RESOURCE_NAME}"
+		rm   ${HOME}/.local/bin/${RESOURCE_NAME}				
+		
+		echo "Installation directory and it's contents including this script can be removed"
+		read -p "Delete directory ${SCRIPT_PATH}? (YES for the affirmative)" YN
+		if [ -z $YN ]
+		then
+			echo exiting; exit
+		elif [ $YN = YES ]
+		then
+			echo "Removing Directory ${SCRIPT_PATH}"
+			cd ..
+			rm -rf ${SCRIPT_PATH}
+		fi
+	fi
+
+###### Perform local user only installation	
+	if [ $MODE = U ]
+	then
+		
+		LIB_PATH=${HOME}/.local/lib/${RESOURCE_NAME}
+		EXE_PATH=${HOME}/.local/bin
+			
+		if [ $YN = Y ] || [ $YN = y ]
+		then
+			mkdir -p $LIB_PATH
+			echo -n "Copying files... "
+			cp -rf * ${HOME}/.local/lib/${RESOURCE_NAME}
+			echo "Files copied...\n"
+		fi
+		
+		mkdir -p "${HOME}/.local/bin"
+		rm -f ${HOME}/.local/bin/${RESOURCE_NAME}
+		ln -s ${HOME}/.local/lib/${RESOURCE_NAME}/arduino-ide ${HOME}/.local/bin/${RESOURCE_NAME}
+	
+		# Create a temp dir accessible by all users
+		TMP_DIR=`mktemp --directory`
+
+		# Create *.desktop file using the existing template file
+		sed -e "s,<BINARY_LOCATION>,${EXE_PATH}/${RESOURCE_NAME},g" \
+	    -e "s,<id>,local,g" \
+        -e "s,<ICON_NAME>,${LIB_PATH}/arduino2.png,g" "${SCRIPT_PATH}/desktop.template" > "${TMP_DIR}/${RESOURCE_NAME}l.desktop"
```

Comment:
instead of using `sed` you should use `destkop-file-edit`. It's way easier to understand and exactly designed for that. There is also `desktop-file-install` to install it.